Makefile.kokkos

```diff
@@ -30,7 +30,7 @@ KOKKOS_TRIBITS ?= "no"
 KOKKOS_STANDALONE_CMAKE ?= "no"
 
 # Default settings specific options.
-# Options: force_uvm,use_ldg,rdc,enable_lambda,enable_constexpr
+# Options: force_uvm,use_ldg,rdc,enable_lambda,enable_constexpr,disable_malloc_async
```

Comment:
why are you making this an optional to "disable" it? 
I would argue that for consistency wit the the `Kokkos_ENABLE_IMPL_CUDA_MALLOC_ASYNC` we should call this `enable_malloc_async` so that the user decided to enable it (not enabling the disabling )

Reply:
Because our internal logic just check for existence of the substrings so you can't really do `disable_malloc_async=0`.  And even if we did parse that properly I find it awkward to use.